### PR TITLE
mobile: added functionality to search text

### PR DESCRIPTION
**Pull request by amitsin6h**
"Mobile: Added search text option"

image

Additional note:
```
As note.js screen has SelectDateTimeDialog
so using listening to onChangeText from TextInput of searchTextDialog was having issues
and to overcome used onFocus, onBlur, and this.searchPopupDialog.dismiss callback

What it does is whenever the searchTextDialog is active it will make searchTextDialog unactive and when the searchTextDialog unactive it will make searchTextDialog active
```
```
note-body-viewer.js 
added forceUpdate whenever it receives props and so that search text works realtime
```

I have also added comments in code for better understanding.
Please review and let me know.

### CODE REVIEW

**Comment by tessus** [comment]
Please install the pre-commit hook. Your code has linter issues.

File: ReactNativeClient/lib/components/note-body-viewer.js [hunk 1]
```diff
@@ -39,6 +39,56 @@ class NoteBodyViewer extends Component {
...
+
+		// If the length of the body has changed, then it's something other than a checkbox that has changed,
+		// for example a resource that has been attached to the note while in View mode. In that case, update.
+		return (`${safeGetNoteProp(this.props, 'body')}`).length !== (`${safeGetNoteProp(nextProps, 'body')}`).length;
+	}
+
+	rebuildMd() {
+		// this.mdToHtml_.clearCache();
+		this.forceUpdate();
+	}
```
Comment (laurent22) [inline ReactNativeClient/lib/components/note-body-viewer.js]: Where did the new code above come from? This is something I wrote so I'm surprised it appears as new here. Or is it something I've deleted and you reimported? If so, why?

**Comment by tessus** [comment]
@amitsin6h did you see my comment (re linter) and Laurent's question?

**Comment by amitsin6h** [comment]
@tessus yeah I have seen your comments actually their is some problem with the PR I was asking with Laurent's how I can fix I guess I have to send the PR again. 

Also could you please tell me how I can use pre-commit hook, Is their any configurations that is required

**Comment by tessus** [comment]
@amitsin6h ok, because I haven't seen any questions to Laurent.

> Also could you please tell me how I can use pre-commit hook, Is their any configurations that is required

You only have to install it. Run `npm install` in the top directory. That should do it.

**Comment by amitsin6h** [comment]
@tessus Thanks

**Comment by laurent22** [comment]
> Where did the new code above come from? This is something I wrote so I'm surprised it appears as new here. Or is it something I've deleted and you reimported? If so, why?

Not sure why that was marked as "resolved". That needs to be fixed before it can be reviewed as only your code should be in this PR.

**Comment by amitsin6h** [comment]
@laurent22 the problem was with my fork I didn't update the latest code from upstream branch and just sent the PR due to which it had merge conflict so will I will close this PR and send new PR with the changes.

**Comment by tessus** [comment]
@amitsin6h thanks.

**Pull request closed without merging**